src/graphapi/find_engine.cpp

```diff
@@ -36,12 +36,163 @@
+        assert(add1);
+
+        auto* add2 =
+            dynamic_cast<OperationPointwise*>(graph.findOutNeighByName(add1, "OP_POINTWISE:ADD"));
+        assert(add2);
+
+        return isBiasNode(add1) || isBiasNode(add2);
+    }
+
+public:
+    static std::unique_ptr<GraphPatternMatcher> Make()
+    {
+        return std::make_unique<ConvBiasResAddActive_Fwd_Pattern>();
+    }
+
+    std::string_view name() const final
+    {
+        static const char* n = "convbiasresaddactivation_fwd";
+        return n;
+    }
+
+    bool matches(const OpGraph* graph_ptr) const final
+    {
+        assert(graph_ptr);
+
+        if(!isIsomorphic(*graph_ptr, getPatternGraph()))
+        {
+            return false;
+        }
+
+        return hasBiasNode(*graph_ptr);
+    }
+
+    std::vector<Engine> getEngines(OpGraph* graph_ptr) const override
+    {
+        assert(graph_ptr);
+        assert(matches(graph_ptr));
+        auto& graph = *graph_ptr;
+
+        auto* conv = dynamic_cast<OperationConvolutionForward*>(
+            graph.findOutNeighByName(graph.getSourceNode(), "OP_CONVOLUTION_FORWARD"));
+        assert(conv);
+
+        std::size_t in_c  = conv->getX()->GetLengths()[1];
+        std::size_t wei_c = conv->getW()->GetLengths()[1];
+
+        if(wei_c == std::size_t{0})
+        {
+            MIOPEN_THROW(
+                miopenStatusBadParm,
+                "invalid weight tensor provided for graph matching ConvBiasResAddActive pattern");
+        }
+        else if(in_c % wei_c != std::size_t{0})
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "invalid group count from input and weight tensor for graph matching "
+                         "ConvBiasResAddActive pattern");
+        }
+
+        int groupCount = in_c / wei_c;
+
+        auto* add1 =
+            dynamic_cast<OperationPointwise*>(graph.findOutNeighByName(conv, "OP_POINTWISE:ADD"));
+        assert(add1);
+
+        auto* add2 =
+            dynamic_cast<OperationPointwise*>(graph.findOutNeighByName(add1, "OP_POINTWISE:ADD"));
+        assert(add2);
+
+        auto* activ = dynamic_cast<OperationPointwise*>(
+            graph.findOutNeighByName(add2, "OP_POINTWISE:RELU_FWD"));
+        assert(activ);
+
+        auto [add, bias] =
+            isBiasNode(add1) ? std::make_pair(add2, add1) : std::make_pair(add1, add2);
+
+        bool addXIsVirtual = add->getX()->isVirtual();
+        assert(addXIsVirtual || add->getB()->isVirtual());
+
+        auto convertToFloat = [](auto&& arg) { return static_cast<float>(arg); };
+        // The virtual tensor for the add is the result of the convolution, and combining the
+        // alpha1's allow for users to specify the alpha on either, or both nodes, and have it be
+        // correct.
+        float alpha1 =
+            conv->getAlpha() * (addXIsVirtual ? std::visit(convertToFloat, add->getAlpha1())
```

Comment:
This swaps alpha1 and alpha2 when X is concrete (i.e. B is virtual). Just a wild guess, but it seems counterintuitive. Also, I find this code to be (subjectively ofc) easier to read:
```suggestion
            auto alpha1 = std::visit(convertToFloat, add->getAlpha1();
            auto alpha2 = std::visit(convertToFloat, add->getAlpha2();
            if(addXIsVirtual) std::swap(alpha1, alpha2);
```